modules/extension/ysld/src/test/java/org/geotools/ysld/encode/YsldEncodeTest.java

```diff
@@ -1320,4 +1322,12 @@ public void testSymbolizerUoMOther() throws Exception {
         exception.expect(IllegalArgumentException.class);
         Ysld.encode(sld(fts), out);
     }
+
+    @Test
+    public void testColorObject() throws Exception {
+        RasterSymbolizerEncoder encoder = new RasterSymbolizerEncoder(null);
+        Literal literal = new FilterFactoryImpl().literal(Color.BLACK);
+        Object obj = encoder.toObjOrNull(literal);
+        assertEquals(obj.getClass(), Color.class);
+    }
```

Comment:
With the change suggested above, this test will no longer be valid. Something that tests a full style would probably be better anyway; something like this should work (using the style example reported in the ticket):

```
    @Test
    public void testEncodeColorMapEntry() throws IOException {
        StyledLayerDescriptor style = new YsldParser(new ByteArrayInputStream(
                    ("name:  Test\n" +
                    "title: Test Style title\n" +
                    "abstract: Styling of Test layer\n" +
                    "feature-styles:\n" +
                    "- rules:\n" +
                    "  - title: raster\n" +
                    "    symbolizers:\n" +
                    "      - raster:\n" +
                    "          opacity: 1.0\n" +
                    "          color-map:\n" +
                    "            type: values\n" +
                    "            entries:\n" +
                    "            - ['#e20374', 1.0, 1, Lorem Ipsum (magenta = covered)]").getBytes())).parse();

        RasterSymbolizer symbolizer = (RasterSymbolizer)((NamedLayer)style.getStyledLayers()[0]).styles().get(0).featureTypeStyles().get(0).rules().get(0).symbolizers().get(0);

        ColorMap colorMap = symbolizer.getColorMap();
        RasterSymbolizerEncoder.ColorMapEntryIterator iterator = new RasterSymbolizerEncoder(symbolizer).new ColorMapEntryIterator(colorMap);
        Tuple map = iterator.next();
        assertEquals("('#E20374',1.0,1,Lorem Ipsum (magenta = covered))", map.toString());
    }
```